src/citrine/_serialization/properties.py

```diff
@@ -408,7 +408,8 @@ def _deserialize(self, value) -> datetime:
         raise TypeError("{} must be an int or a string".format(value))
 
     def _serialize(self, value: datetime) -> int:
-        return int(arrow.get(value).float_timestamp * 1000)
+        # Add 100 nanoseconds to avoid floating point truncation issues from microseconds
```

Comment:
What are the truncation issues to which you're referring?

Reply:
As per the test I added, 269815509154 was being changed to 269815509153 on round trip de-serialization.  Multiplying by 1000 brought finite precision issues into the least significant spot, and `int` would then just miss when truncating.